rust/connlib/tunnel/src/client.rs

```diff
@@ -436,32 +438,42 @@ impl ClientState {
             return None;
         };
 
-        // We read this here to prevent problems with the borrow checker
-        let is_dns_resource = self.is_dns_resource(&resource);
-
-        let Some(peer) = peer_by_resource_mut(&self.resources_gateways, &mut self.peers, resource)
-        else {
-            self.on_not_connected_resource(resource, &dst, now);
-            return None;
-        };
-
         let packet = maybe_mangle_dns_query_to_cidr_resource(
             packet,
             &self.dns_mapping,
             &mut self.mangled_dns_queries,
             now,
         );
```

Comment:
This now needs to happen before `on_not_connected_resource` to ensure we buffer the correctly mangled packet.